src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> lastNode = last;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        last = newNode;
+        if (first == null) {
+            first = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkForIndexEqualSize(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> actual = getNode(index);
+            Node<T> newNode = new Node<>(null, value, actual);
+            actual.prev = newNode;
+            first = newNode;
+            size++;
```

Comment:
As an option: this part can be brought into a separate method, which will allow not to overload the current method. can simplify the readability and size of the code. This can make the method easier to read

Reply:
This one i will leave for mentor's decision.